pkg/routeagent/controllers/route/route.go

```diff
@@ -548,6 +611,18 @@ func (r *Controller) processNextEndpoint() bool {
 				klog.Fatalf("Unable to create VxLAN interface on gateway node (%s): %v", hostname, err)
 			}
 
+			err = r.configureIPRule(AddRoute)
+			if err != nil {
+				klog.Errorf("Unable to add ip rule to table %d on Gateway node %s: %v",
+					RouteAgentHostNetworkTableID, hostname, err)
+			}
+
+			if r.localCableDriver == "" {
+				r.localCableDriver = endpoint.Spec.Backend
+			}
```

Comment:
Should this also be checked/set when on a non-gateway node as well?

Reply:
For non-gateway we don't use that later, it wouldn't hurt to do always if we did.

Reply:
Sorry not sure  follow.  `localCableDriver` is used by `configureRoute` which is called by `updateRoutingRulesForHostNetworkSupport ` on all endpoint-processing paths.

Reply:
Basically, we need the cableDriver used in the cluster. Although currently we do not support a mix and match of cableDrivers, its a good idea to only depend on what is the localDriver on the cluster. This information, as you pointed out, will be used on the GatewayNode to program routing rules to remoteSubnets.

On nonGateway nodes, we call the API updateRoutingRulesForHostNetworkSupport with FlushRouteTable which does not require the driver information (i.e., driver specific interface).

Reply:
BTW we are doing this before the controller is initialized in L227, so no need to do it again on non-gateway nodes.